### ISSUE TITLE
No warnings when implicitly declaring a function.

### ISSUE THREAD

**Issue by ghost** [microsoft/vscode-cpptools#9127]
Bug type: Language Service

**Describe the bug**
- OS and Version: Windows 10 Pro 21H2, Build 19044.1526
- VS Code Version: 1.65.2 (system-wide installer)
- C/C++ Extension Version: v1.8.4
- Other extensions you installed (and if the issue persists after disabling them):
"Remote - WSL" (C/C++ extension is installed in WSL)
"One Dark Pro" (persists)
"Trailing Spaces" (persists)
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

I expect the extension to display warning squiggles and show a warning in the "problems" panel when a function is implicitly declared. However, this is not the case, and nothing is shown to indicate there is a function being called with no declaration. This is undesired as I often realize only after running the build task and having it interrupted by the compiler, thereby wasting time. The issue occurs even in new, empty workspaces.

---

**Steps to reproduce**
1. Create a new C file
2. Implicitly declare a function
3. You will observe that the extension has no issues with this, and makes no effort to tell you.

---

**Code sample and logs**
- Please provide code sample, your c_cpp_properties.json and logs.
No ``c_cpp_properties.json``. However, here are the relevant options in my ``settings.json``:
<details>
<summary>settings.json</summary>

```json
{
    "C_Cpp.default.compilerPath": "/usr/bin/x86_64-w64-mingw32-gcc",
    "C_Cpp.default.cStandard": "c99",
    "C_Cpp.default.intelliSenseMode": "windows-gcc-x64",
    "C_Cpp.clang_format_fallbackStyle": "{BasedOnStyle: Google, IndentWidth: 4, ColumnLimit: 0, SortIncludes: Never}",
    "C_Cpp.autocompleteAddParentheses": true,
    "C_Cpp.autoAddFileAssociations": false,
}
```

</details>

- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette

<details>
<summary>logs</summary>

```
-------- Diagnostics - 4/1/2022, 7:30:28 PM
Version: 1.8.4
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [],
    "cppStandard": "gnu++17",
    "compilerArgs": [],
    "cStandard": "c99",
    "intelliSenseMode": "windows-gcc-x64",
    "intelliSenseModeIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": false,
    "mergeConfigurations": false,
    "compilerPath": "/usr/bin/x86_64-w64-mingw32-gcc",
    "compilerPathIsExplicit": true,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ /home/sherlock/bug/bug.c ]:
    /home/sherlock/bug/bug.c
Translation Unit Configurations:
[ /home/sherlock/bug/bug.c ]:
    Process ID: 316
    Memory Usage: 13 MB
    Compiler Path: /usr/bin/x86_64-w64-mingw32-gcc
    Includes:
        /usr/lib/gcc/x86_64-w64-mingw32/11.2.0/include
        /usr/lib/gcc/x86_64-w64-mingw32/11.2.0/include-fixed
        /usr/x86_64-w64-mingw32/include
    Standard Version: c99
    IntelliSense Mode: windows-gcc-x64
    Other Flags:
        --gcc
        --gnu_version=110200
Total Memory Usage: 13 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 2480
```

</details>

- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)
Nothing relevant.

---

**Screenshots**

![image](https://user-images.githubusercontent.com/80807073/161305106-e4973025-6c23-4a48-b7a9-b3b1e437c4a1.png)

**Comment by sean-mcmanus** [comment]
You can get a warning from clang-tidy via using the Run Code Analysis feature:
![image](https://user-images.githubusercontent.com/19859882/161311205-51d22311-97b9-4519-848c-070a8e924b91.png)

For warnings from IntelliSense, this is a duplicate of https://github.com/microsoft/vscode-cpptools/issues/828.
